File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import Counter
def solution(participant, completion):
    answer = ''
    apart = Counter(participant)
    acom = Counter(completion)

    #apart for문 돌면서, 같은거 - 하기
    for key, value in apart.items() :
        if apart[key] != acom[key] :
            answer = key
    return answer
        

File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String, Integer> apart = new HashMap<>();
        Map<String, Integer> acom = new HashMap<>();

        for (String p : participant) {
            apart.put(p, apart.getOrDefault(p, 0) + 1);
        }

        for (String c : completion) {
            acom.put(c, acom.getOrDefault(c, 0) + 1);
        }

        for (String key : apart.keySet()) {
            if (!apart.get(key).equals(acom.get(key))) {
                answer = key;
                break;
            }
        }

        return answer;
    }
}
